src/plugins/vis_type_vega/public/data_model/vega_parser.ts

```diff
@@ -405,6 +405,10 @@ The URL is an identifier only. OpenSearch Dashboards and your browser will never
             )
           );
         }
+        // Converting the visibleVisLayers array back to a map
```

Comment:
Should we also modify the unit test in the src/plugins/vis_type_vega/public/data_model/vega_parser.test.js to test the logic?

I am thinking maybe for this part:
```
describe('VegaParser._parseConfig', () => {
  function check(spec, expectedConfig, expectedSpec, warnCount) {
    return async () => {
      expectedSpec = expectedSpec || cloneDeep(spec);
      const vp = new VegaParser(spec);
      const config = await vp._parseConfig();
      expect(config).toEqual(expectedConfig);
      expect(vp.spec).toEqual(expectedSpec);
      expect(vp.warnings).toHaveLength(warnCount || 0);
    };
  }

  test('no config', check({}, {}, {}));
  test('simple config', check({ config: { a: 1 } }, {}));
  test(
    'opensearchDashboards config',
    check({ config: { kibana: { a: 1 } } }, { a: 1 }, { config: {} })
  );
  test('_hostConfig', check({ _hostConfig: { a: 1 } }, { a: 1 }, {}, 1));
});
```

We could add some configs 
```
const spec = { config: { kibana: {visibleVisLayers: [['layerA', true],['layerB',false]]}}};
const expectedConfig = {
    visibleVisLayers: new Map([
      ['layerA', true],
      ['layerB', false],
    ]),
  };
```
Then also add a test
```
test(
    'visibleVisLayers conversion to map',
   spec, expectedConfig, { config: {} }, 0)
  );
```

Reply:
Great idea! I'll add a test case here.